### PR TITLE
Class list detail

### PR DESCRIPTION
**Pull request by ClaireGray**
Student view their class attendance works now.
Teacher able to view their student's class attendance works now.
Created 2 separate view for class index works now.
Student no longer have access to creating a class, creating a lecture and managing student.
Remove unnecessary lines of code in lecture index.

### CODE REVIEW

File: UniWatch/Views/Class/Index.cshtml [hunk 1]
```diff
@@ -1,54 +1,117 @@
 ﻿@model IEnumerable<UniWatch.Models.Class>
 
-<h3>My Classes</h3>
 
-<p>
-    @Html.ActionLink("New Class", "Create", null, new { @class = "btn btn-success" })
-</p>
-
-@if(!Model.Any())
+@if (Request.IsAuthenticated && User.IsInRole("Student"))
```
Comment (the-pat) [inline UniWatch/Views/Class/Index.cshtml:4]: The controller should take care of user authentication, right?

File: UniWatch/Views/Class/Index.cshtml [hunk 2]
```diff
@@ -1,54 +1,117 @@
...
+    {
+        @:No class displayed. You are currently not enrolled in any class.
+    }
+    if (!Model.Any())
+    {
+        @:It's empty in here. Create a new class.
+    }
+    else
+    {
+        <div class="table-responsive">
```
Comment (the-pat) [inline UniWatch/Views/Class/Index.cshtml:18]: This section for the student (lines 5-61) seem to duplicate much of the code in the teacher section of the code (lines 62-116). Is there a way to minimize the duplication?

File: UniWatch/Views/Class/Index.cshtml [hunk 3]
```diff
@@ -1,54 +1,117 @@
...
-                    @Html.DisplayNameFor(model => model.Year)
-                </th>
-                <th></th>
-            </tr>
-            @foreach(var item in Model)
-            {
+    <h3>Courses Taught</h3>
+
+    <p>
+        @Html.ActionLink("New Class", "Create", null, new { @class = "btn btn-success" })
```
Comment (the-pat) [inline UniWatch/Views/Class/Index.cshtml:67]: Place another new class button underneath this table as well.

File: UniWatch/Views/Lecture/Index.cshtml [hunk 4]
```diff
@@ -7,13 +7,67 @@
 <h2>@Model.ClassName</h2>
 <hr />
 
-@{
+@if (Request.IsAuthenticated && User.IsInRole("Student"))
```
Comment (the-pat) [inline UniWatch/Views/Lecture/Index.cshtml:10]: The controller should take care of user authentication, right?

File: UniWatch/Views/Lecture/Index.cshtml [hunk 5]
```diff
@@ -7,13 +7,67 @@
...
+{
```
Comment (the-pat) [inline UniWatch/Views/Lecture/Index.cshtml:11]: Similar to the other view, there is a lot of code duplication between the student and teacher view. Can this be minimized?

**Review by the-pat — CHANGES_REQUESTED** [review]
Overall the changes look good. There seems to be some logic in the View that is handled in the Controller (checking if the user is authenticated...correct me if I understand that part incorrectly) and a lot of duplicated code in the Viewe between the Student and Teacher sections.

Also, merge the `master` branch into this branch to prevent merge conflicts.

File: UniWatch/Views/Lecture/Index.cshtml (hunk 5 above)
Comment (frankibem) [reply UniWatch/Views/Lecture/Index.cshtml:11]: I largely agree with @the-pat's comments. You should not need to check for authentication or role in the veiw. A different route will be to have one index view file for teachers and another for students. Then you just need to return the appropriate one from the controller

**Comment by frankibem** [comment]
Is this branch still being maintained? If not, delete it.

**Comment by ClaireGray** [comment]
ok. I will delete this branch frank

**Pull request closed without merging**